File: hex2dec.js
// hex2dec
let x = "1A2";
let a = 0;

for (let i = 0; i < x.length; i++) {
  let h = x.length - 1 - i;
  let dec;

  if (x[i] === "A" || x[i] === "a") dec = 10;
  else if (x[i] === "B" || x[i] === "b") dec = 11;
  else if (x[i] === "C" || x[i] === "c") dec = 12;
  else if (x[i] === "D" || x[i] === "d") dec = 13;
  else if (x[i] === "E" || x[i] === "e") dec = 14;
  else if (x[i] === "F" || x[i] === "f") dec = 15;
  else dec = Number(x[i]);

  a += dec * (16 ** h);
  console.log("szorzat:", x[i], "érték:", dec, "hatvány:", h, "összeg:", a);
}

console.log("A decimális szám:", a);

File: hex2dec.py
#hex2dec
x="1A2"
dec=0
a=0

for i in range (len(x)):
    h=len(x)-1-i
    print ("A szorzat: ", x[i])
    print ("hatvány",h)

    if x[i]=="A" or "a":
        dec=10
    elif x[i]=="B" or "b":
        dec=11
    elif x[i]=="C" or "c":
        dec=12
    elif x[i]=="D" or "d":
        dec=13
    elif x[i]=="E" or "e":
        dec=14
    elif x[i]=="F" or "f":
        dec=15
    else:
        dec=x[i]
    
    a+=dec*(16*h)
    print ("A hatvány összege:" , 16**h)
#    print (a)

print ("A decimális szám értéke:", a)
